CRM/Pcpteams/Utils.php

```diff
@@ -763,4 +763,29 @@ static function checkPcpType($pcpId) {
     return in_array('Team', $aContactTypes) ? 'Team' : 'Indiviual';
     
   }
+  
+  static function adjustTeamTarget($pcpId) {
+     if(empty($pcpId)) {
+      return NULL;
+    }
+    $pcpType  = CRM_Pcpteams_Utils::checkPcpType($pcpId);
+    // only for indiviual pcp
+    if ($pcpType != CRM_Pcpteams_Constant::C_CONTACT_SUB_TYPE_TEAM) {
+      $result = civicrm_api('pcpteams', 'get', array( 'version' => 3, 'pcp_id' => $pcpId,));
+      foreach ($result['values'] as $pcp => $value) {
+        if(!empty($value['team_pcp_id']) && !empty($value['goal_amount'])) {
+          // get team Admin for this team
+          $teamAdmin = CRM_Pcpteams_Utils::getTeamAdmin($value['team_pcp_id']);
+          // make user the team admin updates goal amount
+          if($teamAdmin == $value['contact_id']) {
+            $query = "UPDATE civicrm_pcp SET goal_amount = %1 WHERE (goal_amount is NULL OR goal_amount = 0)";
+            $queryParams = array(
+              1 => array($value['goal_amount'], 'String'),
+            );
+            CRM_Core_DAO::executeQuery($query, $queryParams);
+          }
+        }
+      }
+    }
+  }
```

Comment:
We 'll need to clean it a bit. Its not using permissin layer, 2. Can be simplified using a query